jpc/src/lib.rs

```diff
@@ -285,29 +285,30 @@ impl CodingStyleDefault {
     fn new(value: u8) -> Vec<CodingStyleDefault> {
         let mut coding_styles: Vec<CodingStyleDefault> = vec![];
 
-        if value & 0b_0000_0001 > 0 {
+        if value & 0b1001 == 0 {
```

Comment:
I think we should only check the LSB here (and corresponding bits for SOP and EPH). There are uses of the higher bits in 15444-2 (see Table A.5) - they won't always be set to zero.